### PR TITLE
(maint) unify log messages with other providers

### PR DESCRIPTION
**Pull request by ciprianbadescu**
downgrade warning to debug messages, as they are not warnings for
 our users

 also, fixed one code review remark that got somehow overlooked

### CODE REVIEW

File: lib/puppet/provider/package/apt.rb [hunk 1]
```diff
@@ -56,19 +56,15 @@ def best_version(should_range)
       is = line.split('|')[1].strip
       begin
         is_version = DebianVersion.parse(is)
-        available_versions << is_version
-      rescue DebianVersion::ValidationFailure
-        Puppet.warning("Cannot parse #{is} as a debian version")
+        available_versions << is_version if should_range.include?(is_version)
+      rescue DebianVersion::ValidationFailure
```
Comment (luchihoratiu) [inline lib/puppet/provider/package/apt.rb]: A space character got away at the end of the line

File: lib/puppet/provider/package/apt.rb [hunk 2]
```diff
@@ -56,19 +56,15 @@ def best_version(should_range)
...
+        Puppet.debug("Cannot parse #{is} as a debian version")
       end
     end
 
-    included_available_version = available_versions.select do |version|
-      should_range.include?(version)
-    end
-
-    return included_available_version[-1] unless included_available_version.empty?
+    return available_versions.to_a.last unless available_versions.empty?
```
Comment (gimmyxd) [inline lib/puppet/provider/package/apt.rb]: you don'' need `.to_a` as `available_versions` is an array

Comment (ciprianbadescu) [reply lib/puppet/provider/package/apt.rb]: I tried something, but not working
```
2.6.4 :001 > a=SortedSet.new
 => #<SortedSet: {}> 
2.6.4 :002 > a << 3
 => #<SortedSet: {3}> 
2.6.4 :003 > a[-1]
Traceback (most recent call last):
        4: from /usr/local/rvm/rubies/ruby-2.6.4/bin/irb:23:in `<main>'
        3: from /usr/local/rvm/rubies/ruby-2.6.4/bin/irb:23:in `load'
        2: from /usr/local/rvm/rubies/ruby-2.6.4/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        1: from (irb):3
NoMethodError (undefined method `[]' for #<SortedSet: {3}>)
2.6.4 :004 > a.last
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.6.4/bin/irb:23:in `<main>'
        4: from /usr/local/rvm/rubies/ruby-2.6.4/bin/irb:23:in `load'
        3: from /usr/local/rvm/rubies/ruby-2.6.4/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):4
        1: from (irb):4:in `rescue in irb_binding'
NoMethodError (undefined method `last' for #<SortedSet: {3}>)
2.6.4 :005 > a.to_a.last
 => 3 
```

Comment (gimmyxd) [reply lib/puppet/provider/package/apt.rb]: Oh, you're using a sorted set, i missed that. Not sure which is more efficient, `sorted_set.to_a.last` or `array.sort.last`.

Comment (ciprianbadescu) [reply lib/puppet/provider/package/apt.rb]: seems `a.max` should be the right way

**Pull request merged**